### PR TITLE
Update cmake minimum version to 2.8.12

### PR DESCRIPTION
**Pull request by brawner**
In the most recent release of cmake (3.19), they have started deprecating cmake versions < `2.8.12`. 

Example cmake warnings
https://ci.ros2.org/view/nightly/job/nightly_win_rel/1760/cmake/

Build all of ros2, testing up to `rcutils`
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13157/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8088/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10872/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13162/)

Signed-off-by: Stephen Brawner <brawner@gmail.com>

### CODE REVIEW

**Comment by sloretz** [comment]
CI https://github.com/ros/urdfdom_headers/pull/67#issuecomment-733042583

**Comment by brawner** [comment]
Rerun ci jobs
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13161/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8092/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10876/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13167/)

It doesn't look like python_orocos_kdl is actually built in ros2. Closing.

**Review by sloretz — APPROVED** [review]

**Comment by brawner** [comment]
Actually reopening because I missed `orocos_kdl` itself here. I'll rerun the ci jobs shortly.

File: python_orocos_kdl/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.4.6)
+cmake_minimum_required(VERSION 2.8.12)
```
Comment (clalancette) [inline python_orocos_kdl/CMakeLists.txt]: I think we can skip this one; it is `AMENT_IGNORE`, and I'm certain we'd have to do more complicated things to make it work anyway.

**Review by clalancette — CHANGES_REQUESTED** [review]

File: python_orocos_kdl/CMakeLists.txt (hunk 1 above)
Comment (brawner) [reply python_orocos_kdl/CMakeLists.txt]: Realized you meant this change, not the PR in total. Removed this specific change

**Review by clalancette — APPROVED** [review]
Looks good with green CI.

**Comment by brawner** [comment]
Testing `--packages-select orocos_kdl`
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13179/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8110/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10895/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13193/)

**Pull request merged**